File: src/main/java/me/pljr/pljrapispigot/objects/PLJRScoreboard.java
package me.pljr.pljrapispigot.objects;

import me.pljr.pljrapispigot.utils.ScoreboardUtil;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.scoreboard.DisplaySlot;
import org.bukkit.scoreboard.Objective;
import org.bukkit.scoreboard.Score;
import org.bukkit.scoreboard.Scoreboard;

import java.util.List;

public class PLJRScoreboard {
    private final String title;
    private final List<String> lines;

    public PLJRScoreboard(PLJRScoreboard scoreboard){
        this.title = scoreboard.getTitle();
        this.lines = scoreboard.getLines();
    }

    public PLJRScoreboard(String title, List<String> lines){
        this.title = title;
        this.lines = lines;
    }

    public String getTitle() {
        return title;
    }

    public List<String> getLines() {
        return lines;
    }

    public Scoreboard parseScoreboard(){
        Scoreboard scoreboard = Bukkit.getScoreboardManager().getNewScoreboard();

        Objective objective = scoreboard.registerNewObjective(title, "");
        objective.setDisplaySlot(DisplaySlot.SIDEBAR);
        objective.setDisplayName(title);

        int slot = 16;
        for (String line : lines){
            Score score = objective.getScore(line);
            score.setScore(slot);
            slot--;
        }

        return scoreboard;
    }

    public void send(Player player){
        ScoreboardUtil.send(player, this);
    }

    public void broadcast(){
        ScoreboardUtil.broadcast(this);
    }
}

File: src/main/kotlin/me/pljr/pljrapispigot/scoreboard/PLJRScoreboard.kt
package me.pljr.pljrapispigot.scoreboard

import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.scoreboard.DisplaySlot
import org.bukkit.scoreboard.Scoreboard

class PLJRScoreboard(val title: String, val lines: MutableList<String>) {

    fun toScoreboard() : Scoreboard {
        val scoreBoard = Bukkit.getScoreboardManager().newScoreboard

        val objective = scoreBoard.registerNewObjective(title, "", title)
        objective.displaySlot = DisplaySlot.SIDEBAR

        var slot = 16
        lines.forEach {
            objective.getScore(it).score = slot
            slot--
        }

        return scoreBoard
    }

    fun send(player: Player) = sendScoreBoard(player, this)

    fun broadcast() = Bukkit.getOnlinePlayers().forEach { send(it) }
}